homeassistant/components/recorder/statistics.py

```diff
@@ -1248,48 +1265,40 @@ def _get_oldest_sum_statistic_in_sub_period(
             else:
                 period = start_time.replace(minute=0, second=0, microsecond=0)
             prev_period = period - table.duration
-            stmt += lambda q: q.filter(table.start == prev_period)
+            stmt += lambda q: q.filter(table.start >= prev_period)
         stats = execute_stmt_lambda_element(session, stmt)
-        return (
-            (stats[0].sum, process_timestamp(stats[0].start)) if stats else (None, None)
-        )
+        return stats[0].sum if stats else None
 
-    oldest_start: datetime | None
     oldest_sum: float | None = None
 
+    if not tail_only and oldest_stat is not None:
+        assert main_start_time is not None
+        period = main_start_time.replace(minute=0, second=0, microsecond=0)
+        prev_period = period - Statistics.duration
+        if prev_period < oldest_stat:
+            return 0
+
     if head_start_time is not None:
-        oldest_sum, oldest_start = _get_oldest_sum_statistic_in_sub_period(
+        oldest_sum = _get_oldest_sum_statistic_in_sub_period(
             session, head_start_time, StatisticsShortTerm, metadata_id
         )
-        if (
-            oldest_start is not None
-            and oldest_start < head_start_time
-            and oldest_sum is not None
-        ):
+        if oldest_sum is not None:
             return oldest_sum
 
     if not tail_only:
         assert main_start_time is not None
-        oldest_sum, oldest_start = _get_oldest_sum_statistic_in_sub_period(
+        oldest_sum = _get_oldest_sum_statistic_in_sub_period(
             session, main_start_time, Statistics, metadata_id
         )
-        if (
-            oldest_start is not None
-            and oldest_start < main_start_time
-            and oldest_sum is not None
-        ):
+        if oldest_sum is not None:
             return oldest_sum
         return 0
 
     if tail_start_time is not None:
-        oldest_sum, oldest_start = _get_oldest_sum_statistic_in_sub_period(
+        oldest_sum = _get_oldest_sum_statistic_in_sub_period(
             session, tail_start_time, StatisticsShortTerm, metadata_id
         )
-        if (
-            oldest_start is not None
-            and oldest_start < tail_start_time
-            and oldest_sum is not None
-        ):
+        if oldest_sum is not None:
```

Comment:
- Wallrus :=
- Can be combined into a single `if` at that point the wallrus is in place.